native/src/main/java/io/ballerina/lib/avro/deserialize/visitor/DeserializeVisitor.java

```diff
@@ -0,0 +1,452 @@
+        }
+        return (double) data;
+    }
+
+    public BArray visitBytes(Object data) {
+        return ValueCreator.createArrayValue(((ByteBuffer) data).array());
+    }
+
+    public BArray visitFixed(Object data) {
+        GenericData.Fixed fixed = (GenericData.Fixed) data;
+        return ValueCreator.createArrayValue(fixed.bytes());
+    }
+
+    public BString visitString(Object data) {
+        return fromString(data.toString());
+    }
+
+    public static Type extractMapType(Type type) throws Exception {
+        Type mapType = type;
+        if (type.getTag() != TypeTags.RECORD_TYPE_TAG) {
+            throw new Exception("Type is not a record type.");
+        }
+        for (Map.Entry<String, Field> entry : ((RecordType) type).getFields().entrySet()) {
+            Field fieldValue = entry.getValue();
+            if (fieldValue != null) {
+                Type fieldType = fieldValue.getFieldType();
+                switch (fieldType.getTag()) {
+                    case TypeTags.MAP_TAG ->
+                            mapType = fieldType;
+                    case TypeTags.INTERSECTION_TAG -> {
+                        Type referredType = getMutableType((IntersectionType) fieldType);
+                        if (referredType.getTag() == TypeTags.MAP_TAG) {
+                            mapType = referredType;
+                        }
+                    }
+                    default -> {
+                        Type referType = TypeUtils.getReferredType(fieldType);
+                        if (referType.getTag() == TypeTags.MAP_TAG) {
+                            mapType = referType;
+                        }
+                    }
+                }
+            }
+        }
+        return mapType;
+    }
+
+    public static RecordType extractRecordType(RecordType type) {
+        Map<String, Field> fieldsMap = type.getFields();
+        RecordType recType = type;
+        for (Map.Entry<String, Field> entry : fieldsMap.entrySet()) {
+            Field fieldValue = entry.getValue();
+            if (fieldValue != null) {
+                Type fieldType = fieldValue.getFieldType();
+                switch (fieldType.getTag()) {
+                    case TypeTags.RECORD_TYPE_TAG ->
+                            recType = (RecordType) fieldType;
+                    case TypeTags.INTERSECTION_TAG -> {
+                        Type getType = getMutableType(fieldType);
+                        if (getType.getTag() == TypeTags.RECORD_TYPE_TAG) {
+                            recType = (RecordType) getType;
+                        }
+                    }
+                    default -> {
+                        Type referredType = TypeUtils.getReferredType(fieldType);
+                        if (referredType.getTag() == TypeTags.RECORD_TYPE_TAG) {
+                            recType = (RecordType) referredType;
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Consider breaking the large if-else ladders and also use proper names. The `getType` sounds more like a method name than a variable name. 

Plus, don't we need `break` statements with each case here?

Reply:
Fixed the variable name. The `break` statement is not necessary when using `switch` with the `arrow` syntax.